_opensearch/rest-api/document-apis/update-document.md

```diff
@@ -87,6 +87,35 @@ You can also use a script to tell OpenSearch how to update your document.
 }
 ```
 
+### Upsert
+
+Upsert is an operation that conditionally updates an existing document or inserts a new one based on information in the object. In the sample below, the `upsert` object updates the name and adds the `age` field if a document already exists. If a document does not exist, a new one is indexed using content in the `upsert` object.
+
+```json
+{
+  "doc": {
+    "first_name": "Thomas",
+    "last_name": "Wayne"
```

Comment:
Should the `last_name` is the "doc" body vs the `last_name` in the "upsert" body match? If not, why?

Reply:
The idea in this example is that if the last name changes, the upsert updates the new information in an existing doc or adds (inserts) a new doc if it doesn't already exist. I used a completely different last name in the example to try making it clearer. I also worked some language in the description to help. Does that make better sense?

Reply:
Changed name completely to help make example explicit.